### PR TITLE
Adds draw_quadrangle method

### PR DESCRIPTION
**Pull request by jeffreypaul15**
Adds `draw_quadrangle` from sunpy. I've taken the docstring from sunpy itself as it doesn't change here.
I don't think we need to worry about the `kwargs` which are passed to astropy's `Quadrangle` do we?
If yes, then we need to have a separate `quad_kwargs` or something.

image

### CODE REVIEW

**Comment by jeffreypaul15** [comment]
Oops, forgot to rebase.

**Comment by nabobalis** [comment]
How does the example look?

**Comment by jeffreypaul15** [comment]
> How does the example look?

I've updated the PR description

File: sunkit_pyvista/plotter.py [hunk 1]
```diff
@@ -174,6 +175,36 @@ def plot_solar_axis(self, length=2.5, arrow_kwargs={}, **kwargs):
...
+            have shape ``(2,)`` to simultaneously define ``top_right``.
+        top_right : `~astropy.coordinates.SkyCoord`
+            The top-right coordinate of the quadrangle.
+        width : `astropy.units.Quantity`, optional
+            The width of the quadrangle. Required if ``top_right`` is omitted.
+        height : `astropy.units.Quantity`
+            The height of the quadrangle. Required if ``top_right`` is omitted.
+        **kwargs : Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
+        """
+        quadrangle_patch = m.draw_quadrangle(bottom_left=bottom_left, top_right=top_right,
```
Comment (dstansby) [inline sunkit_pyvista/plotter.py:199]: This is not ideal, as we don't want to have a map as an input argument. I propose that instead we just copy the logic from `draw_quadrangle` directly into this function to get the patch:
```python
bottom_left, top_right = get_rectangle_coordinates(
            bottom_left, top_right=top_right, width=width, height=height)

        width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
        height = top_right.spherical.lat - bottom_left.spherical.lat

        kwergs = {
            "transform": axes.get_transform(bottom_left.frame.replicate_without_data()),
            "edgecolor": "white",
            "fill": False,
        }
        kwergs.update(kwargs)
        quad = Quadrangle(self._get_lon_lat(bottom_left), width, height, **kwergs)
```

Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:199]: We would need to have `get_rectangle_coordinages()`and `Quadrangle`  imported then right?
Also, `self` is a reference to a map object, how do we handle that?

Comment (dstansby) [reply sunkit_pyvista/plotter.py:199]: Yep, we would need those imports. For _get_lon_lat, I think you can replace it with
```python
bottom_left = bottom_left.spherical
quad = Quadrangle((bottom_left.lon, bottom_left.lat), width, height, **kwergs)
```

Comment (jeffreypaul15) [reply sunkit_pyvista/plotter.py:199]: Sure, thanks!

**Review by dstansby — APPROVED** [review]

**Pull request merged**